10 lab

EkaterinaPushkareva04 opened the pull request.

EkaterinaPushkareva04 committed:
Add files via upload

EkaterinaPushkareva04 committed:
Add files via upload

NemirovichMark commented:
Здравствуйте, Екатерина.

С диаграммами разобраться. Как обозначаются агрегация, реализация, наследование (всегда от зависящего к зависимому)
![image](https://github.com/NemirovichMark/BBI-23-2/assets/112826355/7b55bd51-ae91-405e-af74-ccab42f27a54)
![image](https://github.com/NemirovichMark/BBI-23-2/assets/112826355/ac1cb00b-2b20-4837-b486-6c527f344579)
На диаграмме состояний не должно быть действий в блоках. Там должны быть состояния: пустой массив или файл или заполненный и т п. А вот над стрелочками действия пишутся.
![image](https://github.com/NemirovichMark/BBI-23-2/assets/112826355/31341aac-65e4-4751-933f-6a52b6ab6740)

Выделил то, чего не хватает в выводе.
Прочтите данные из файлов raw_data.xml, **data.json и data.xml** и выведите на консоль. 

Самостоятельно написать счетчик (отдельным методом) и использовать его вместо:
 return Students.Count(s => s.AverageMark == mark);
if(a.Students.Count(s => s.AverageMark == i) <
    b.Students.Count(s => s.AverageMark == i))
return Students.Count(s => s.AverageMark >= from && s.AverageMark <= to);

Доп. компоненты делать не будете, да?

На доработку.

NemirovichMark commented:
Свою сортировку также сделать
  Students = Students.OrderBy(s => s.AverageMark).ToList();

EkaterinaPushkareva04 committed:
10 lab fix

NemirovichMark commented:
Здравствуйте, Екатерина.

Группа в себе содержит студентов, а не наследуется от них.
![image](https://github.com/NemirovichMark/BBI-23-2/assets/112826355/1393c83f-e806-41ae-ab22-d9eac89fc4a7)

Без Linq, через циклы считать

    int ICountable.Count(double mark)
    {
        return Students.Count(s => s.AverageMark == mark);
    }
    int ICountable.Percentage(double from, double to)
    {
        return Students.Count(s => s.AverageMark >= from && s.AverageMark <= to);
    }
}
 string IReportable.GenerateReport()
 {
     double averageMarkGroup = Students.Count > 0 ? Students.Sum(s => s.AverageMark) / Students.Count : 0;
     double averageMisses = Students.Count > 0 ? Students.Sum(s => s.Misses) / Students.Count : 0;
     int failingStudents = Students.Count(s => s.AverageMark < 3.5);
     double failingPercentage = Students.Count > 0 ? ((double)failingStudents / Students.Count) * 100 : 0;

     return $"Group: {Name}, Number of Students: {Students.Count}, Average Mark: {averageMarkGroup}, Average Misses per Student: {averageMisses}, Number of Failing Students: {failingStudents}, Percentage of Failing Students: {failingPercentage}%.";
 }

На доработку.

EkaterinaPushkareva04 committed:
10 lab ready

Добрый вечер, скидываю Вам исправленный код + диаграмму классов, проверьте, пожалуйста.

NemirovichMark commented:
Отлично. Принял. Остальные члены команды тоже пусть присылают работу.
